firebase-firestore/src/main/java/com/google/firebase/firestore/local/LocalStore.java

```diff
@@ -606,6 +606,20 @@ public void notifyLocalViewChanges(List<LocalViewChanges> viewChanges) {
               TargetData updatedTargetData =
                   targetData.withLastLimboFreeSnapshotVersion(lastLimboFreeSnapshotVersion);
               queryDataByTarget.put(targetId, updatedTargetData);
+
```

Comment:
@wu-hui It seemed weird to create a fake TargetChange to pass into shouldPersistTargetData, so I just did the check here. It does work as expected. If you want to still make the changes to use shouldPersistTargetData, feel free.

Reply:
Can you pass a null as TargetChange? Then in shouldPersistTargetData, you can do an early return for the limboFreeSnapshotVersion check, then check for null on TargetChange, together with the changes > 0 check.

I'll merge your PR first, then do a followup change myself if you prefer. Thank you!

Reply:
Hadn't thought of passing null, that could work.
I'll let you merge this and make the follow up changes, I am swamped with other work.
Thanks!